app/controllers/users_controller.rb

```diff
@@ -0,0 +1,109 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :set_skill_levels, except: [:index]
+  before_action :set_user_positions, except: [:index]
+  before_action :set_all_users, except: [:index]
+
+  # GET /users
+  # GET /users.json
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  # GET /users/1.json
+  def show
+    @user_skills = Skill.where user: @user
+  end
+
+  # GET /users/new
+  def new
+    @user = User.new
+    @users = User.all.pluck(:full_name).to_json
+  end
+
+  # GET /users/1/edit
+  def edit
+    @users = User.all.pluck(:full_name).to_json
+  end
+
+  # POST /users
+  # POST /users.json
+  def create
+    supervisor = User.find_by full_name: params[:user][:supervisor_name]
+    params[:user][:supervisor_id] = supervisor.id if supervisor
+    @user = User.new(user_params)
+
+    respond_to do |format|
+      if @user.save
+        format.html { redirect_to @user, notice: 'User was successfully created.' }
+        format.json { render :show, status: :created, location: @user }
+      else
+        format.html { render :new }
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /users/1
+  # PATCH/PUT /users/1.json
+  def update
+    # todo: improvements here? need the text field for the edit but don't know if i can do away with it
+    # update subordinate's supervisor_name fields
+    if params[:user][:full_name] != @user.full_name
+      @user.subordinates.each do |subordinate|
+        subordinate.supervisor_name = params[:user][:full_name]
```

Comment:
within each user's form page, there is a field where users enter their supervisor's name. when a user (who may be a supervisor) updates their name, this block updates their subordinate's supervisor names.